protostar/upgrader/upgrade_toml.py

```diff
@@ -0,0 +1,65 @@
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Optional
+
+import tomli
+import tomli_w
+
+from protostar.utils.protostar_directory import (
+    ProtostarDirectory,
+    VersionManager,
+    VersionType,
+)
+
+
+@dataclass(frozen=True)
+class UpgradeTOML:
+    """
+    Pythonic representation of the upgrade.toml.
```

Comment:
I think it is a bit cryptic name for a class and suggests it is an upgrade configuration. `LatestVersionCacheTOML`? WDYT?

Reply:
Maybe `protostar_cache.toml`? IDK I am open for ideas just `upgrade.toml` feels bad